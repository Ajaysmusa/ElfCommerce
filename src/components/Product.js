import React, { Component } from 'react';
import '../App.css';
import {
  Row, Col,
} from 'reactstrap';


class Product extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="no-border product-item">
        <a href={this.props.productURL}>
          <div className="product-cover" style={{ backgroundImage: `url(${this.props.productImage})` }} />
        </a>
        <br />
        <Row className="product-item-footer">
          <Col md={8} xs={8}>
            <a href={this.props.productURL}>{this.props.productName}</a>
          </Col>
          <Col md={4} xs={4} className="text-right">
            <strong>{this.props.productPrice}</strong>
          </Col>
        </Row>
      </div>
    );
  }
}

export default Product;
