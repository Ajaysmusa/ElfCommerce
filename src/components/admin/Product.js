import React, { Component } from 'react';
import { Container, Row, Col, Button } from 'reactstrap';
import { Link } from 'react-router-dom';

class Product extends Component {
  render() {
    return (
      <Container>
        <h3>Products</h3>
        <Breadcrumb>
          <Breadcrumb.Item href="/admin/dashboard">Dashboard</Breadcrumb.Item>
          <Breadcrumb.Item active>Products</Breadcrumb.Item>
        </Breadcrumb>
        <Row>
          <Col md={12}>
            <ListGroup>
              <ListGroupItem>
                <Row>
                  <Col md={2}>Nicholas Chen</Col>
                  <Col md={8}>nicholas_chan82@hotmail.com</Col>
                  <Col md={2}>Delete</Col>
                </Row>
              </ListGroupItem>
              <ListGroupItem>
                <Row>
                  <Col md={2}>Nicholas Chen</Col>
                  <Col md={8}>nicholas_chan82@hotmail.com</Col>
                  <Col md={2}>Delete</Col>
                </Row>
              </ListGroupItem>
            </ListGroup>
          </Col>
        </Row>
        <Button bsStyle="primary" style={{ float: 'right' }}>
          Checkout
        </Button>
      </Container>
    );
  }
}

export default Product;
