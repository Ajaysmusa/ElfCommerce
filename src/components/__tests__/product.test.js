import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import Product from "../product";

Enzyme.configure({ adapter: new Adapter() });

test("product component with display a product image, name and price", () => {
  const product = shallow(
    <Product
      productName="Test product"
      productURL="products/test-product-1"
      productImage="http://assets.teenvogue.com/photos/57dfbfe5046b3a2e2a7364bd/master/pass/Kylie-Armytee.jpg"
      productPrice="$12.90"
    />
  );
  const json = toJSON(product);
  expect(json).toMatchSnapshot();
});
