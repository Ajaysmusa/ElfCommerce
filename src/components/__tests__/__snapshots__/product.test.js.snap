// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product component with display a product image, name and price 1`] = `
<div
  className="no-border product-item"
>
  <a
    href="products/test-product-1"
  >
    <div
      className="product-cover"
      style={
        Object {
          "backgroundImage": "url(http://assets.teenvogue.com/photos/57dfbfe5046b3a2e2a7364bd/master/pass/Kylie-Armytee.jpg)",
        }
      }
    />
  </a>
  <br />
  <Row
    className="product-item-footer"
    tag="div"
  >
    <Col
      md={8}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs={8}
    >
      <a
        href="products/test-product-1"
      >
        Test product
      </a>
    </Col>
    <Col
      className="text-right"
      md={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs={4}
    >
      <strong>
        $12.90
      </strong>
    </Col>
  </Row>
</div>
`;
